### PR TITLE
Renovation: window refactor

### PR DESCRIPTION
**Pull request by LazyLahtak**
Get rid of using `this` in `window` (allow named export for all functions)

### CODE REVIEW

File: js/core/utils/window.js [hunk 1]
```diff
@@ -2,49 +2,40 @@
...
+    const windowWidth = domAdapter.getDocumentElement()['clientWidth'];
+
+    return screenFactorFunc(windowWidth);
+};
+
+const getNavigator = () => hasWindow() ? windowObject.navigator : { userAgent: '' };
+
+module.exports = {
+    hasWindow,
+    getWindow,
```
Comment (MaximKudriavtsev) [inline js/core/utils/window.js:36]: Why aren't we using `export const ...` ?

Comment (LazyLahtak) [reply js/core/utils/window.js:36]: In some places we use
```js
const windowUtils = require('../core/utils/window');
```
This is incompatible with
```js
export const hasWindow = ...;
export const getWindow = ...;
```
or with
```js
export {
    hasWindow,
    getWindow,
    ...
}
```

If we want to use it, we should get rid of all `windowUtils = require`. Should we do that in this PR?

Comment (MaximKudriavtsev) [reply js/core/utils/window.js:36]: I think we want to do it.

> Should we do that in this PR?

Yes, we can.

**Review by churkin — APPROVED** [review]

File: js/core/utils/window.js (hunk 1 above)
Comment (LazyLahtak) [reply js/core/utils/window.js:36]: We have too many places, where we use `require`. It's quite difficult to change them all in same time (many dependencies between components).

**Review by MaximKudriavtsev — APPROVED** [review]

**Review by nightskylark — APPROVED** [review]

**Pull request merged**